File: 이지수/Python/Programmers_42889.py
def solution(N, stages):
    levels = [0] * (N + 1)
    for stage in stages :
        levels[stage-1] += 1
    n = sum(levels)
    levels = levels[:-1]
    failures = []
    for count in levels :
        if count == 0 : failures.append(0)
        else : failures.append(float(count)/float(n))
        n -= count
    failures = list(sorted(sorted(enumerate(failures), key=lambda x : x[0]), key=lambda x : x[1], reverse=True))
    return [index+1 for index, _ in failures]

File: 이지수/Swift/Programmers_42889.swift
import Foundation

func solution(_ N:Int, _ stages:[Int]) -> [Int] {
    var levels: [Int] = [Int](repeating: 0, count: N + 1)
    for stage in stages {
        levels[stage-1] += 1
    }
    var failure: [Double] = []
    var n = levels.reduce(0) {$0 + $1}
    levels.removeLast()
    for count in levels {
        failure.append(Double(count)/Double(n))
        n -= count
    }
    return failure.enumerated().sorted() {
        if $0.1 > $1.1 { return true }
        else { return $0.0 < $1.0 }
    }.map{$0.0 + 1}
}
